### PR TITLE
Moving ideal methods to category of rings

### PR DESCRIPTION
**Pull request by fchapoton**
This is moving ``unit_ideal``, ``zero_ideal``, ``principal_ideal`` from the Ring class to the category of rings

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/rings/quotient_ring.py [hunk 1]
```diff
@@ -1474,10 +1474,14 @@ def _macaulay2_init_(self, macaulay2=None):
...
         return I.ring()._operator('/', I)
 
-    def _ideal_class_(self, num_gens):
+    def _ideal_class_(self, n=0):
         r"""
         Use a specialized class for quotient ring ideals.
 
+        INPUT:
+
+        - ``n`` -- optional integer, the number of generators
```
Comment (dcoudert) [inline src/sage/rings/quotient_ring.py]: shouldn't it be
```py
- ``n`` -- integer (default: ``0``); the number of generators
```

File: src/sage/rings/tate_algebra.py [hunk 2]
```diff
@@ -889,13 +889,13 @@ def _pushout_(self, R):
...
 
-    def _ideal_class_(self, n):
+    def _ideal_class_(self, n=0):
         r"""
         Return the class that handles ideals in this Tate algebra.
 
         INPUT:
 
-        - ``n`` -- number of generators
+        - ``n`` -- optional integer, number of generators
```
Comment (dcoudert) [inline src/sage/rings/tate_algebra.py]: same here

**Review by dcoudert — APPROVED** [review]
otherwise LGTM.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by dcoudert** [comment]
I'm not sure what you did with the last commit as I see more than 2570 modified files...

**Comment by fchapoton** [comment]
indeed. I hade merged, which is not what github likes. Now rebased, thanks

**Comment by fchapoton** [comment]
as the rebase was trivial (over the doc patchbomb), I set back to positive

**Pull request merged**